hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactSplit.java

```diff
@@ -315,11 +315,13 @@ public void switchCompaction(boolean onOrOff) {
         LOG.info("Re-Initializing compactions because user switched on compactions");
         reInitializeCompactionsExecutors();
       }
-    } else {
-      LOG.info("Interrupting running compactions because user switched off compactions");
-      interrupt();
+      setCompactionsEnabled(onOrOff);
```

Comment:
Let's also revert this change? It does not change anything actually? Only a style change?

Reply:
Oh, here I think it is not just a style change, for disable compaction, I think the order we would better to set the flag first and then shut down the thread pool, in the reverse order of operations when compaction is enabled.